### ISSUE TITLE
IT-047-BACKEND: Generate skill slug automatically from skill name

### ISSUE THREAD

**Issue by Fehmi-Yildirim** [Fehmi-Yildirim/it-talent-backend#50]
##IT-047-BACKEND: Generate skill slug automatically from skill name 
## Goal

Move skill slug generation to the backend so that the backend is the single source of truth for skill slugs.

The frontend should only send the skill name, category, and description. The backend must generate and maintain the skill slug consistently for both API operations and database seeding.

## Expected behavior

A skill slug must be generated automatically from the skill name.

Examples:

* `React` → `react`
* `React Native` → `react-native`
* `Machine Learning` → `machine-learning`
* `Node.js` → `node-js`
* `Next.js` → `next-js`
* `C#` → `c`
* `Full-stack development` → `full-stack-development`

Slug generation should:

* convert characters to lowercase
* normalize accented characters where applicable
* replace spaces with `-`
* remove unsupported/special characters
* collapse repeated separators
* remove leading/trailing separators
* produce a stable, URL-friendly slug

## API changes

Update the skill create/update API so that clients no longer need to provide `slug`.

### Create

The API should accept:

```json
{
  "name": "React Native",
  "category": "MOBILE",
  "description": "Cross-platform mobile development"
}
```

The backend generates:

```text
react-native
```

### Update

When the skill name changes, the backend should generate the corresponding slug from the new name.

The API should remain responsible for enforcing slug uniqueness.

## Centralized slug generation

Implement slug generation in one reusable backend location.

The same implementation must be used by:

* skill creation
* skill updates
* database seeding

Do not duplicate slug-generation logic between the service and Prisma seed.

## Duplicate / unique slug handling

The existing unique-slug behavior must remain intact.

If two skills would result in the same slug, the backend must handle the conflict according to the existing application/database rules.

Do not silently overwrite an existing skill.

## Prisma seed

Move the standard skill data into the backend seed.

The seed should contain the standard skill list and generate the slug from each skill name using the same centralized slug-generation logic as the API.

The seed should be idempotent so it can safely be executed multiple times without creating duplicate skills.

### Standard skill categories

The initial seed should contain these 84 skills:

#### Frontend

* React
* Vue.js
* Angular
* JavaScript
* TypeScript
* HTML
* CSS
* Tailwind CSS

#### Backend

* Node.js
* Java
* C#
* Python
* PHP
* Go
* Ruby
* .NET
* REST API
* GraphQL

#### Fullstack

* MERN
* MEAN
* Next.js
* Full-stack development

#### Mobile

* React Native
* Flutter
* Swift
* Kotlin
* Android
* iOS

#### DevOps

* Docker
* Kubernetes
* CI/CD
* Jenkins
* GitHub Actions
* Terraform
* Ansible

#### Cloud

* AWS
* Azure
* Google Cloud
* Firebase

#### Data

* Data Analysis
* Pandas
* NumPy
* Power BI
* Tableau
* Apache Spark

#### AI / ML

* Machine Learning
* Deep Learning
* TensorFlow
* PyTorch
* NLP
* Generative AI

#### Security

* Cybersecurity
* Penetration Testing
* OWASP
* Network Security
* Identity & Access Management

#### Database

* PostgreSQL
* MySQL
* MongoDB
* Redis
* SQL
* NoSQL

#### Testing

* Jest
* Vitest
* Cypress
* Playwright
* Selenium
* Unit Testing
* Integration Testing

#### Project Management

* Agile
* Scrum
* Kanban
* Jira
* Project Management

#### Design

* UI Design
* UX Design
* Figma
* Adobe XD
* Responsive Design

#### Other

* Git
* GitHub
* Linux
* Bash
* Technical Writing

## Tests

Add or update backend tests covering:

* slug generation
* lowercase conversion
* whitespace normalization
* special-character handling
* repeated separators
* accented characters
* skill creation without a supplied slug
* skill update with a changed name
* duplicate slug handling
* seed behavior
* idempotent seed execution

## Acceptance criteria

* [ ] Backend generates skill slugs automatically from skill names.
* [ ] Clients no longer need to send `slug` when creating a skill.
* [ ] Clients no longer need to send `slug` when updating a skill.
* [ ] Create and update use the same centralized slug-generation logic.
* [ ] Prisma seed uses the same slug-generation logic.
* [ ] All 84 standard skills are included in the backend seed.
* [ ] Seed execution is idempotent.
* [ ] Existing unique slug constraints continue to work.
* [ ] Existing skills are not unintentionally broken.
* [ ] Backend tests cover the new behavior.
* [ ] All existing backend tests pass.
* [ ] Lint passes.
* [ ] Build passes.
